app/js/components/login/Login.jsx

```diff
@@ -19,9 +21,21 @@ class Login extends Component {
     this.state = {
       username: '',
       password: '',
+      loaded: false,
+      formSubmitted: false,
     };
   }
 
+  componentDidMount() {
+    this.checkAuthentication();
+  }
+
+  checkAuthentication = () => {
+    getLoginStatus('/session')
+      .then(response => this.handleResponse(response))
+      .catch(error => this.handleError(error));
+  }
```

Comment:
this should be in an `actionCreator` and not inside the component

Reply:
Thanks @Efosaok, its a plain react application hence no actionCreators.